internal/lightcone/abundance/timewaitsfornoone/timewaitsfornoone.go

```diff
@@ -0,0 +1,113 @@
+package timewaitsfornoone
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/equip/lightcone"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	time key.Modifier = "time-waits-for-no-one"
+	// key for extra dmg attack, not for modifier.
+	extraDmg key.Attack = "time-waits-for-no-one-extra-damage"
+)
+
+type healRecorder struct {
+	cooldown    int
```

Comment:
cooldown as an `int` is not clear. Given how it is currently used, would recommend changing to a bool and having it make more sense in usage (only perform an attack if `cooldown == false`